pkg/fleet/installer/service/apm_sockets.go

```diff
@@ -0,0 +1,153 @@
+		log.Warn("Failed to unmarshal agent configuration, using default installer sockets")
+		return apmSocket, statsdSocket, nil
+	}
+	if cfg.ApmSocketConfig.ReceiverSocket != "" {
+		apmSocket = cfg.ApmSocketConfig.ReceiverSocket
+	}
+	if cfg.DogstatsdSocket != "" {
+		statsdSocket = cfg.DogstatsdSocket
+	}
+	return apmSocket, statsdSocket, nil
+}
+
+// configureSocketsEnv configures the sockets for the agent & injector
+func configureSocketsEnv() error {
+	envFile := newFileMutator(envFilePath, setSocketEnvs, nil, nil)
+	defer envFile.cleanup()
+	rollback, err := envFile.mutate()
+	if err != nil {
+		rollbackErr := rollback()
+		if rollbackErr != nil {
+			log.Warnf("Failed to rollback environment file: %v", rollbackErr)
+		}
+		return fmt.Errorf("error configuring sockets: %w", err)
+	}
+	// Make sure the file is word readable
+	if err := os.Chmod(envFilePath, 0644); err != nil {
+		return fmt.Errorf("error changing permissions of %s: %w", envFilePath, err)
+	}
+	return nil
+}
+
+// setSocketEnvs sets the socket environment variables
+func setSocketEnvs(envFile []byte) ([]byte, error) {
+	apmSocket, statsdSocket, err := getSocketsPath()
+	if err != nil {
+		return nil, fmt.Errorf("error getting sockets path: %w", err)
+	}
+	envs := map[string]string{
+		"DD_APM_RECEIVER_SOCKET": apmSocket,
+		"DD_DOGSTATSD_SOCKET":    statsdSocket,
+		"DD_USE_DOGSTATSD":       "true",
+	}
+	return addEnvsIfNotSet(envs, envFile)
+}
+
+// addEnvsIfNotSet adds environment variables to the environment file if they are not already set
+func addEnvsIfNotSet(envs map[string]string, envFile []byte) ([]byte, error) {
+	// Build a map of the existing env vars
+	existingEnvs := map[string]bool{}
+	for _, line := range strings.Split(string(envFile), "\n") {
+		parts := strings.SplitN(line, "=", 2)
+		if len(parts) < 2 {
+			continue
+		}
+		existingEnvs[strings.TrimSpace(parts[0])] = true
+	}
+
+	var buffer bytes.Buffer
+	buffer.Write(envFile)
+	if len(envFile) > 0 && envFile[len(envFile)-1] != '\n' {
+		buffer.WriteByte('\n')
+	}
+	for key, value := range envs {
+		if !existingEnvs[key] {
+			buffer.WriteString(fmt.Sprintf("%s=%s\n", key, value))
+		}
+	}
+	return buffer.Bytes(), nil
+}
+
+// addSystemDEnvOverrides adds /etc/datadog-agent/environment variables to the defined systemd units
+// The unit should contain the .service suffix (e.g. datadog-agent-exp.service)
+//
+// Reloading systemd & restarting the unit has to be done separately by the caller
+func addSystemDEnvOverrides(unit string) error {
+	// The - is important as it lets the unit start even if the file is missing.
+	content := []byte(fmt.Sprintf("[Service]\nEnvironmentFile=-%s\n", envFilePath))
+
+	// We don't need a file mutator here as we're fully hard coding the content.
+	// We don't really need to remove the file either as it'll just be ignored once the
+	// unit is removed.
+	path := fmt.Sprintf("%s/%s.d/datadog_environment.conf", systemdPath, unit)
```

Comment:
take the habit of using `filepath.Join`, probably no impact here but it's not great to do this by hand